vite-project/src/strategypage/RadarChart.tsx

```diff
@@ -0,0 +1,220 @@
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+
+  const polygonPoints: point[] = new Array(inputs.length);
+  const statPoints: point[] = new Array(inputs.length);
+
+  //function that draws the outer regular polygos
+
+  function drawOuterPolygon(
+    polygonRadius: number,
+    canvas: HTMLCanvasElement,
+    polygonPoints: point[],
+    stokeColor: string
+  ): void {
+    const context = canvas.getContext("2d");
+    const canvasRadius: number = canvas.width / 2;
+    if (context) {
+      context.beginPath();
+
+      for (let i = 0; i < inputs.length + 1; i++) {
+        const currentX: number =
+          polygonRadius * Math.sin((i * Math.PI * 2) / inputs.length) +
+          canvasRadius;
+        const currentY: number =
+          polygonRadius * Math.cos((i * Math.PI * 2) / inputs.length) +
+          canvasRadius;
+
+        context.lineTo(currentX, currentY);
+        context.lineTo(canvasRadius, canvasRadius);
+        context.lineTo(currentX, currentY);
+
+        const currentPoint: point = { x: currentX, y: currentY };
+
+        polygonPoints[i] = currentPoint;
+      }
+      context.closePath();
+      context.strokeStyle = stokeColor;
+      context.stroke();
+    }
+  }
+
+  //function that draws the inner regular polygos
+
+  function drawInnerPolygon(
+    polygonRadius: number,
+    canvas: HTMLCanvasElement,
+    stokeColor: string
+  ): void {
+    const context = canvas.getContext("2d");
+    const canvasRadius: number = canvas.width / 2;
+    if (context) {
+      context.beginPath();
+
+      for (let i = 0; i < inputs.length + 1; i++) {
+        const currentX: number =
+          polygonRadius * Math.sin((i * Math.PI * 2) / inputs.length) +
+          canvasRadius;
+        const currentY: number =
+          polygonRadius * Math.cos((i * Math.PI * 2) / inputs.length) +
+          canvasRadius;
+
+        context.lineTo(currentX, currentY);
+
+        const currentPoint: point = { x: currentX, y: currentY };
+
+        polygonPoints[i] = currentPoint;
+      }
+      context.closePath();
+      context.strokeStyle = stokeColor;
+      context.stroke();
+    }
+  }
+
+  //function that puts lables in a regular shape placement
+
+  function drawLables(polygonRadius: number, canvas: HTMLCanvasElement): void {
+    const context = canvas.getContext("2d");
+    const canvasRadius: number = canvas.width / 2;
+
+    for (let i = 0; i < inputs.length; i++) {
+      const currentX: number =
+        polygonRadius * Math.sin((i * Math.PI * 2) / inputs.length) +
+        canvasRadius;
+      const currentY: number =
+        polygonRadius * Math.cos((i * Math.PI * 2) / inputs.length) +
+        canvasRadius;
+
+      if (context) {
+        context.fillText(inputs[i].name, currentX, currentY);
+      }
+    }
+  }
+
+  useEffect(() => {
+    const canvas = canvasRef.current;
+
+    if (canvas) {
+      const context = canvas.getContext("2d");
+      if (context) {
```

Comment:
pls move your code to not be in these ifs. this is hard to look at. for inspiration look at my code😎